### PR TITLE
Added support for cursor key decoder callback.

### PR DESCRIPTION
**Pull request by ks16**
Added support for cursor key decoder. 

In case when cursor key value must be decoded before querying the database client code may set a custom decoder using the `withKeyDecoder` fluent method. Example with uuid key stored in binary format in the database:
```
CursorPagination::make()->withCursorColumn('uuid')->withKeyDecoder(fn($value) => Uuid::fromString($value)->getBytes());
```

### CODE REVIEW

File: src/CursorPagination.php [hunk 1]
```diff
@@ -266,4 +279,11 @@ private function cursor(array $page): Cursor
         );
     }
 
+    private function decodePaginationKey($value): mixed
+    {
+        $decoder = $this->keyDecoderCallback ?? fn($value) => $value;
+
+        return $decoder($value);
```
Comment (ankorstore-haddowg) [inline src/CursorPagination.php]:
```suggestion
        return $this->keyDecoderCallback ? $this->keyDecoderCallback($value) : $value;
```

File: src/CursorPagination.php [hunk 2]
```diff
@@ -61,6 +62,8 @@ class CursorPagination implements Paginator
      */
     private ?string $primaryKey = null;
 
+    private ?Closure $keyDecoderCallback = null;
```
Comment (ankorstore-haddowg) [inline src/CursorPagination.php:68]: doc block here

File: src/CursorPagination.php [hunk 3]
```diff
@@ -194,6 +197,16 @@ public function withKeyName(string $column): self
         return $this;
     }
 
+    /**
+     * @inheritDoc
+     */
```
Comment (ankorstore-haddowg) [inline src/CursorPagination.php]: missing docbloc

**Review by SofLesc — APPROVED** [review]

File: src/CursorPagination.php (hunk 2 above)
Comment (ankorstore-haddowg) [reply src/CursorPagination.php:68]: and i would use `callable` rather than `Closure` for added flexibility

**Review by ankorstore-haddowg — CHANGES_REQUESTED** [review]

File: src/CursorPagination.php (hunk 2 above)
Comment (ks16) [reply src/CursorPagination.php:68]: `callable` is not supported for class attributes.

File: src/CursorPagination.php (hunk 3 above)
Comment (ks16) [reply src/CursorPagination.php]: Done

File: src/CursorPagination.php (hunk 1 above)
Comment (ks16) [reply src/CursorPagination.php]: Done.

File: src/CursorPagination.php (hunk 2 above)
Comment (SofLesc) [reply src/CursorPagination.php:68]: ❓ What's the doc block for as the property has already a type ?

Comment (ankorstore-haddowg) [reply src/CursorPagination.php:68]: because this is a fork and we want to pr it so need to follow their style 🤷

Comment (SofLesc) [reply src/CursorPagination.php:68]: ok, sure. I thought this was a Ankorstore guideline :)

File: src/CursorPagination.php [hunk 4]
```diff
@@ -194,6 +200,22 @@ public function withKeyName(string $column): self
...
+     *
+     * In case key column value is stored not as a string, for example uuid in binary format,
+     * it has to be decoded before passing it to Cursor.
+     *
+     * @param callable $callback
+     * @return $this
+     */
+    public function withKeyDecoder(callable $callback): self
+    {
+        $this->keyDecoderCallback = $callback;
```
Comment (ankorstore-haddowg) [inline src/CursorPagination.php]:
```suggestion
        $this->keyDecoderCallback = Closure::fromCallable($callback);
```
here or you may get a type error due to the Callable type on the property

Comment (ks16) [reply src/CursorPagination.php]: Done

**Review by ankorstore-haddowg — APPROVED** [review]

**Review by samvaughton — APPROVED** [review]

**Pull request merged**